File: lab2/FileReverse.java
//-----------------------------------------------------------------------------
//Guangyang Chen, 1518931
//gchen34@ucsc.edu
//CMPS12B, Jan 24th
//This is a program that produce a file contains reverse string of the input file
// FileReverse.java
//-----------------------------------------------------------------------------
import java.io.*;
import java.util.Scanner;

class FileReverse{
   public static void main(String[] args) throws IOException{

      Scanner in = null;
      PrintWriter out = null;
      String line = null;
      String[] token = null;
      int i, n, lineNumber = 0;
// check number of command line arguments is at least 2
 if(args.length < 2){
         System.out.println("Usage: FileCopy <input file> <output file>");
         System.exit(1);
      }
// open files
 in = new Scanner(new File(args[0]));
      out = new PrintWriter(new FileWriter(args[1]));
// read lines from in, extract and print tokens from each line
 while( in.hasNextLine() ){
         lineNumber++;
// trim leading and trailing spaces, then add one trailing space so
//  split works on blank lines
 line = in.nextLine().trim() + " "; 
// split line around white space
token = line.split("\\s+");
// print out tokens
n = token.length;
        //out.println("Line " + lineNumber + " contains " + n + " tokens:");
         for(i=0; i<n; i++){
            out.println(stringReverse(token[i],token[i].length()));
   }
      }
 // close files
 in.close();
 out.close();
   }
public static String stringReverse(String s, int n){
if(n>0){
return s.charAt(n-1)+stringReverse(s,n-1);
}
return "";
}

}

File: lab3/FileReverse.c
//------------------------------------------------------------------------------//FileReverse.c
//Reads input file, then reversed and prints each word on a separate line of the// output file
//Guangyang Chen, gchen34@ucsc.edu, 1518931
//Lab 3
//------------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>

//StringReverse method
void stringReverse(char* s){
	int i;
	int j = strlen(s)-1;
	char temp;
	for(i=0; i<j; i++){
	temp=s[i];
	s[i]=s[j];
	s[j]=temp;
	j--;
 }
}

//Main method
int main(int argc, char * argv[]){
   FILE *in, *out;   // handles for input and output files
   char word[256];   // char array to store words from input file
 
   // check command line for correct number of arguments
   if( argc != 3 ){
      printf("Usage: %s <input file> <output file>\n", argv[0]);
      exit(EXIT_FAILURE);
   }

   // open input file for reading 
    in = fopen(argv[1], "r");
   if( in==NULL ){
      printf("Unable to read from file %s\n", argv[1]);
      exit(EXIT_FAILURE);
   }

   // open output file for writing 
   out = fopen(argv[2], "w");
   if( out==NULL ){
      printf("Unable to write to file %s\n", argv[2]);
      exit(EXIT_FAILURE);
   }

   // read words from input file, print on separate lines to output file 
   while(fscanf(in, " %s", word)!=EOF){
        stringReverse(word);
	fprintf(out, "%s\n", word);
   }

   // close input and output files 
   fclose(in);
   fclose(out);
   
   return(EXIT_SUCCESS);
 }
